sisl/viz/plotly/plot.py

```diff
@@ -2438,24 +2435,19 @@ def _get_figure(self, rows, cols, arrange, make_subplot_kwargs):
                 cols = nplots
                 rows = 1
             elif arrange == 'square':
-                cols = np.sqrt(nplots)
-                rows = np.sqrt(nplots)
+                cols = nplots ** 0.5
+                rows = nplots ** 0.5
         elif rows is None:
-            rows = nplots/cols
+            # ensure it is large enough by adding 1 if they don't add up
+            rows = nplots // cols + min(1, nplots % cols)
         elif cols is None:
-            cols = nplots/rows
+            # ensure it is large enough by adding 1 if they don't add up
+            cols = nplots // rows + min(1, nplots % rows)
 
-        if cols % 1 != 0 or rows % 1 != 0:
-            raise ValueError(f'It is impossible to draw a layout with {rows} rows and {cols} cols. ' +
-                             f'Please review the values provided, keeping in mind that the layout should accomodate {nplots} plots')
-        elif cols*rows > nplots:
-            # They will see the empty spaces, not worth it to print something I believe
-            pass
-        elif cols*rows < nplots:
-            print(f'{nplots - cols*rows} plots will be missing, because a {rows}x{cols} layout was requested and you have {nplots}')
+        rows, cols = int(rows), int(cols)
 
-        rows = int(rows)
-        cols = int(cols)
+        if cols * rows < nplots:
+            warn(f'requested {nplots} on a {rows}x{cols} grid layout. {nplots - cols*rows} plots will be missing.")
```

Comment:
Cool!

Reply:
Yeah, I thought this was less error prone ;)